### PR TITLE
Add OpenAICredentials and CompletionModel blocks

### PR DESCRIPTION
**Pull request by ahuang11**
Closes

### Checklist

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-openai/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [x] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [x] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-openai/blob/main/CHANGELOG.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,4 +25,4 @@ Released on ????? ?th, 20??.
 
 ### Added
 
-- `task_name` task - [#1](https://github.com/PrefectHQ/prefect-openai/pull/1)
+- `OpenAICredentials` and `CompletionModel` blocks - [#1](https://github.com/PrefectHQ/prefect-openai/pull/1)
```
Comment (desertaxle) [inline CHANGELOG.md]:
```suggestion
- `OpenAICredentials` and `CompletionModel` blocks - [#2](https://github.com/PrefectHQ/prefect-openai/pull/2)
```

File: prefect_openai/completion.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    A block that contains config for an OpenAI Completion Model.
+
+    Attributes:
+        openai_credentials: The credentials used to authenticate with OpenAI.
+        model: ID of the model to use.
+        temperature: The temperature of the model.
+        max_tokens: The maximum number of tokens to generate.
+        suffix: The suffix to append to the prompt.
+        echo: Whether to echo the prompt.
+        timeout: The maximum time to wait for the model to warm up.
```
Comment (desertaxle) [inline prefect_openai/completion.py:23]: Can we link to some OpenAI docs to help explain what some of these attributes are for?

File: prefect_openai/completion.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    ) -> OpenAIObject:
+        """
+        Create an OpenAI completion given a text prompt. OpenAI will return an object
+        potentially containing multiple `choices`, where the zeroth index is what they
+        consider the "best" completion.
+
+        Args:
+            prompt: The prompt to use for the completion.
+            **acreate_kwargs: Additional keyword arguments to pass
+                to `openai.Completion.acreate`.
```
Comment (desertaxle) [inline prefect_openai/completion.py]: Can we link to the `acreate` docs to guide users on what can be passed in?

File: prefect_openai/completion.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    echo: bool = Field(default=False, description="Whether to echo the prompt.")
+    timeout: Optional[float] = Field(
+        default=None, description="The maximum time to wait for the model to warm up."
+    )
+
+    _block_type_name = "OpenAI Completion Model"
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/QE8JwcbZBmIfiognXDLcY/2bcd4c759f877d37159f576101218b49/open-ai-logo-8B9BFEDC26-seeklogo.com.png?h=250"  # noqa
+
+    @sync_compatible
+    async def create(
```
Comment (desertaxle) [inline prefect_openai/completion.py]: I don't have a strong understanding of OpenAI, but this method name doesn't seem to accurately describe what's going on. My understanding is that it is producing a completion, but `CompletionModel.create` suggests that it's creating a completion model. `create_completion` seems to describe the behavior better, but another option might be better.
```suggestion
    async def create_completion(
```

**Review by desertaxle — CHANGES_REQUESTED** [review]
This is cool! I think that coming up with more real world examples for this collection will be important for exciting users, but can be addressed in another PR.

File: prefect_openai/completion.py (hunk 4 above)
Comment (ahuang11) [reply prefect_openai/completion.py]: create_completion sounds redundant to me: `CompletionModel.create_completion` I'll see if I can find a better method name for it.

Comment (ahuang11) [reply prefect_openai/completion.py]: Thinking of `submit_prompt` or `input_text` from "The best way to start exploring completions is through our Playground. It's simply a text box where you can submit a prompt to generate a completion. To try it yourself, [open this example in Playground](https://beta.openai.com/playground/p/8P6JA6XEx74NTvcRUngWKEYW):

Comment (desertaxle) [reply prefect_openai/completion.py]: I personally like `submit_prompt`

**Comment by ahuang11** [comment]
> This is cool! I think that coming up with more real world examples for this collection will be important for exciting users, but can be addressed in another PR.

I agree. Perhaps I can address this with a landing page like in prefect-dask.

**Review by desertaxle — APPROVED** [review]

**Pull request merged**